source/tutorials/python.rst

```diff
@@ -153,13 +156,38 @@ Finally, you can run the following command to generate all the warnings:
 
 This gives you an idea about how well your code was written, you should have at least 7/10.
 
+Reduce complexity of your code (optional)
+-----------------------------------------
+
+Learning about condition statement, for loop, and various error check is commonly considered a milestone in ones journey to programming mastery.
+However, overusing these tools can make you code complex to understand for anyone, include future you.
+There's a simple way to quickly assess the complexity of your code: look on the left hand side of the code, you want the line formed by the start of the code to be as straight as possible.
+Practically speaking, you want to reduce the use of conditions and loops, and write smaller methods.
+
+.. image:: img/squint-test.jpeg
+    :width: 600px
+
+`Source <https://www.freecodecamp.org/news/how-i-helped-my-partner-learn-to-code-6e1d1953812f/>`_
+
+If you want to gain immediate, further feedback on your python code, ``sourcery`` is a plug-in that gives you immediate feedback on code complexity.
+
+Install this plugin:
+
+.. code:: bash
+
+  code --install-extension sourcery.sourcery
+
+To let sourcery give you feedback, you will need to have some part of the code in a function.
+Hover on a function you wrote and see the quality report on complexity, method length, and working memory.
+`Find the explaination of the metric here <https://github.com/sourcery-ai/sourcery/wiki/Quality-Report#code-metrics>`_.
```

Comment:
I never heard of that before, but a working memory just by ckecking the syntax is quite a good idea. However I am afraid that sometimes (actually most of the time), it is too much variable-dependent.
So maybe adding a note on the fact that this score should not be taken too seriously (maybe pointing to https://sourcery.ai/blog/working-memory/), it will never replace a true benchmark.